src/index.js

```diff
@@ -7,13 +7,29 @@ const POST_FILE = 'POSTFILE'
 const PUT = 'PUT'
 const DELETE = 'DELETE'
 
+
+const JSONParseError = 'JSONParseError'
+const Non200Error = 'Non200Error'
+
 const contentType = {
   GET: 'application/x-www-form-urlencoded',
   POST: 'application/json',
   PUT: 'application/json',
   DELETE: 'application/json',
 }
 
+const networkError = (status, errorType, body, notify, message) => {
+  const errorMessage = message || 'A network error occurred. Please try again.'
+  const e = new Error(errorMessage)
```

Comment:
as in have an actual JSONParseError instead of just returning an error with that code?